.github/CI/spack_setup.sh

```diff
@@ -0,0 +1,56 @@
+# This file should be "sourced" into your environment
+# to set up the spack repository
+
+DIR=${HOME}/spack
+
+#
+# Only update the spack clone once a day
+#
+echo "::group::Spack setup"
+if [ -r ${DIR}/.git/FETCH_HEAD ]; then
+  last_update=`stat -c %Y ${DIR}/.git/FETCH_HEAD`
+  current=`date -d now "+%s"`
+  if [ $(((current - last_update) / 86400)) -gt 1 ]; then
+    echo "Last update ${last_update}, current ${current}: Do git pull spack"
+    cd $DIR && git pull
+  else
+    echo "git pull was less than one day ago"
+  fi
+else
+  echo "git clone spack"
+  git clone https://github.com/spack/spack $DIR || true
+  cd $DIR && git status
+fi
+
+echo "Load spack environment"
+source $DIR/share/spack/setup-env.sh
+
+# Show the known envs
+spack env list | grep ${RUNNER_ENV}
+if [ $? -ne 0 ]; then
```

Comment:
This also doesn't work because the github runner process runs within a shell that will exit automatically on any non-zero return code (like "bash -e").

Reply:
got it, I need to test the return directly with the command.